File: experiment_jp_valid/mutants/for__0__DISCRETE_LOGARITHM_FIND_INTEGER_K_AK_CONGRUENT_MODULO_B.java
static int discreteLogarithm ( int a , int b , int m ) { int n = ( int ) ( Math . sqrt ( m ) + 1 ) ; int an = 1 ; int [ ] value = new int [ m ] ; for ( int i = 1 , cur = an ; i <= n ; ++ i ) { if ( value [ cur ] == 0 ) value [ cur ] = i ; cur = ( cur * an ) % m ; } for ( int i = 0 , cur = b ; i <= n ; ++ i ) { if ( value [ cur ] > 0 ) { int ans = value [ cur ] * n - i ; if ( ans < m ) return ans ; } cur = ( cur * a ) % m ; } return - 1 ; }

File: experiment/java-python/valid/j2pyrawpy/for__0__DISCRETE_LOGARITHM_FIND_INTEGER_K_AK_CONGRUENT_MODULO_B.py
#!/usr/bin/env python
""" generated source for module for__0__DISCRETE_LOGARITHM_FIND_INTEGER_K_AK_CONGRUENT_MODULO_B """
class X(object):
    """ generated source for class X """
    @classmethod
    def discreteLogarithm(cls, a, b, m):
        """ generated source for method discreteLogarithm """
        n = ((Math.sqrt(m) + 1))
        an = 1
        value = [None]*m
        i = 1
        cur = an
        while i <= n:
            if value[cur] == 0:
                value[cur] = i
            cur = (cur * an) % m
            i += 1
        i = 0
        cur = b
        while i <= n:
            if value[cur] > 0:
                if ans < m:
                    return ans
            cur = (cur * a) % m
            i += 1
        return -1

